WordPress/Sniffs/PHP/IniSetSniff.php

```diff
@@ -0,0 +1,173 @@
+			'valid_values' => array( 'true', '1', 'On' ),
+		),
+	);
+
+	/**
+	 * Array of options that are not allowed to be manipulated
+	 *
+	 * @since x.x.x
+	 *
+	 * @var array Multidimensional array with parameter details.
+	 *     $blacklisted_options = array(
+	 *         (string) option name. = array(
+	 *             (string[]) 'invalid_values' = array()
+	 *             (string) 'message'
+	 *         )
+	 *     );
+	 */
+	protected $blacklisted_options = array(
+		'max_execution_time' => array(
+			'message' => 'Use `set_time_limit()` instead.',
+		),
+		'short_open_tag' => array(
+			'invalid_values' => array( 'false', '0', 'Off' ),
+			'message'        => 'Turning off short_open_tag is prohibited as it might possibily break other plugins.',
+		),
+		'bcmath.scale' => array(
+			'message' => 'Use `bcscale()` instead.',
+		),
+		'display_errors' => array(
+			'message' => 'Use `WP_DEBUG_DISPLAY` instead.',
+		),
+		'error_reporting' => array(
+			'message' => 'Use `WP_DEBUG` instead.',
+		),
+		'filter.default' => array(
+			'message' => 'Use the filter flag constants when calling the functions instead (as you will possibly break other plugins if you change this).',
+		),
+		'filter.default_flags' => array(
+			'message' => 'Use the filter flag constants when calling the functions instead (as you will possibly break other plugins if you change this).',
+		),
+		'iconv.input_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'iconv.internal_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'iconv.output_encoding' => array(
+			'message' => 'PHP < 5.6 only - use `iconv_set_encoding()` instead.',
+		),
+		'ignore_user_abort' => array(
+			'message' => 'Use `ignore_user_abort()` instead.',
+		),
+		'log_errors' => array(
+			'message' => 'Use `WP_DEBUG_LOG` instead.',
+		),
+		'memory_limit' => array(
+			'message' => 'Use `wp_raise_memory_limit()` or hook into the filters in that function.',
+		),
+	);
+
+	/**
+	 * Process the parameter of a matched function. Errors if an option
+	 * is found in the blacklist. Warns as 'risky' when the option is not
+	 * found in the whitelist.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$ini_set_function = $this->tokens[ $stackPtr ];
+		$option_name      = $this->strip_quotes( $parameters[1]['raw'] );
+		$option_value     = $this->strip_quotes( $parameters[2]['raw'] );
+		if ( array_key_exists( $option_name, $this->whitelisted_options ) ) {
```

Comment:
Is there any particular reason why you are using the function call `array_key_exists()` rather than the (normally faster) language construct `isset()` ?

Reply:
Started originally as a flat array before adding the 'valid_values' dimension. No particular preference, so will edit this.